assets/js/util/index.js

```diff
@@ -185,20 +172,70 @@ export const readableLargeNumber = ( number, currencyCode = false ) => {
 };
 
 /**
- * Formats a number using the Internationalization Number Format API.
+ * Formats a number with unit using the JS Internationalization Number Format API.
  *
- * @since 1.0.0
- * @see {@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/NumberFormat/NumberFormat|`options` parameter} For all available formatting options.
+ * @since n.e.x.t
  *
- * @param {number} number           The number to format.
- * @param {Object} [options]        Formatting options.
- * @param {string} [options.locale] Locale to use for formatting. Defaults to current locale used by Site Kit.
- * @return {string} The formatted number.
+ * @param {number|string}                     number    The number to format.
+ * @param {(Intl.NumberFormatOptions|string)} [options] Formatting options or unit.
+ * @return {string} The formatted number with unit.
  */
-export const numberFormat = ( number, options = {} ) => {
-	const { locale = getLocale(), ...formatOptions } = options;
+export const numFmt = ( number, options = {} ) => {
+	// Cast parsable values to numeric types.
+	number = isFinite( number ) ? number : Number( number );
+
+	if ( ! isFinite( number ) ) {
+		// eslint-disable-next-line no-console
+		console.warn( 'Invalid number', number, typeof number );
+		return null;
+	}
 
-	return new Intl.NumberFormat( locale, formatOptions ).format( number );
+	let formatOptions = {};
+
+	// Expand shorthand values for units.
+	if ( '%' === options ) {
+		formatOptions = {
+			style: 'percent',
+			maximumFractionDigits: 2,
+		};
+		number /= 100;
+	} else if ( 's' === options ) {
+		formatOptions = {
+			style: 'seconds',
+		};
+	} else if ( 'decimal' === options ) {
+		// Node: `decimal` is our custom style used to return numbers with thousand separators
+		formatOptions = {
+			style: 'decimal',
+		};
```

Comment:
Let's remove this and pass `{ style: 'decimal' }` everywhere it's needed instead. The shorthands here are intended for "units" like %, s etc. Actual styles should be passed in the more appropriate `options` object format.